src/components/formBuilder/formBuilder.jsx

```diff
@@ -0,0 +1,280 @@
+              if (val !== 'type') delete item[val];
+            });
+          }
+          item[name] = value;
+        }
+        return item;
+      });
+
+      setState({ ...state, inputs });
+    } else {
+      setState({ ...state, [name]: value });
+    }
+
+    control();
+  };
+
+  const control = () => {
+    let errors = { inputs: [] };
+    let control = true;
+
+    state.inputs.forEach((item, key) => {
+      if (!item.label) {
+        control = false;
+      } else errors.inputs[key] = {};
+    });
+
+    if (!state.formName) {
+      control = false;
+      errors.formName = 'Form name is required';
+    }
+    if (!state.formDesc) {
+      control = false;
+      errors.formDesc = 'Form description is required';
+    }
+    if (state.inputs.length === 0) {
+      errors.inputList = 'Inputs is required';
+      control = false;
+    }
+    if (control) return true;
+    else setErrors({ ...errors });
+    return false;
+  };
+
+  const onSubmit = () => {
+    if (control()) {
+      props.saveFormSchemaThunk(state);
+      setState({ inputs: [], formName: '', formDesc: '' });
+    }
+  };
+
+  const inputsOptions = key => {
+    if (state.inputs[key].type === 'text' || state.inputs[key].type === 'email')
+      return (
+        <div className={style.input_option_box}>
+          <label htmlFor='placeholder'>Placeholder</label>
+          <input
+            onChange={e => onHandleChnage(e, key)}
+            type='text'
+            id='placeholder'
+            name='placeholder'
+            value={
+              state.inputs[key].placeholder ? state.inputs[key].placeholder : ''
+            }
+          />
+        </div>
+      );
+    else if (state.inputs[key].type === 'number')
+      return (
+        <>
+          <div className={style.input_option_box}>
+            <label htmlFor='maxNumber'>Max number</label>
+            <input
+              onChange={e => onHandleChnage(e, key)}
+              type='number'
+              id='maxNumber'
+              name='maxNumber'
+              value={state.inputs[key].maxNumber}
+            />
+          </div>
+          <div>
+            <label htmlFor='minNumber'>Min number</label>
+            <input
+              onChange={e => onHandleChnage(e, key)}
+              type='number'
+              id='minNumber'
+              name='minNumber'
+              value={state.inputs[key].minNumber}
+            />
+          </div>
+        </>
+      );
+    else if (state.inputs[key].type === 'select')
+      return (
+        <div className={style.input_option_box}>
+          <label htmlFor='selectOption'>Select Option</label>
+          <input
+            onChange={e => onHandleChnage(e, key)}
+            type='text'
+            placeholder='place option ,'
+            id='selectOption'
+            name='selectOption'
+            value={state.inputs[key].selectOption}
+          />
+        </div>
+      );
+    else if (state.inputs[key].type === 'textarea')
```

Comment:
avoid if elses, try to use the same concept I mentioned in the component below, (forms.jsx)
inputs = {
  typeX: ComponentX,
  typeY: ComponentY,
};

const Input = inputs[type];